### DISCUSSION TITLE
GitHub Actions triggers workflow with pull_request_review edited event even if not listening to it, when commenting

### DISCUSSION THREAD

**Discussion by ldeluigi** [community/community#24980] [Actions]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/7/378ab942a2cc99d27ed1a736f9ebf8ae1accdc7f.png" data-download-href="/uploads/short-url/7VltrNt4TaleggxNt7rfs6XNFwP.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/44567586/181087991-5896ebb3-a51f-4305-9d2b-b34e41e2054b.png" alt="image" data-base62-sha1="7VltrNt4TaleggxNt7rfs6XNFwP" width="690" height="328" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/3/7/378ab942a2cc99d27ed1a736f9ebf8ae1accdc7f_2_690x328.png, https://aws1.discourse-cdn.com/github/original/3X/3/7/378ab942a2cc99d27ed1a736f9ebf8ae1accdc7f.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/3/7/378ab942a2cc99d27ed1a736f9ebf8ae1accdc7f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/7/378ab942a2cc99d27ed1a736f9ebf8ae1accdc7f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">927×442 20.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
As you can see, my workflow was triggered with a pull_request_review edited event after I submitted a pull_request_review with comments.
<p>The problem is that in my workflow file I’m listening only to created comments and submitted (not edited) reviews:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/8/a82f3eaed2f6bb8fe9f7ef79d1eded3cba04118b.png" data-download-href="/uploads/short-url/nZPr4yQyRGlSh1c0kVx4dcnJlyP.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/44567586/181087992-c0e1cfd5-a9c0-4e42-8530-d2a138e1df4d.png" alt="image" data-base62-sha1="nZPr4yQyRGlSh1c0kVx4dcnJlyP" width="652" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/8/a82f3eaed2f6bb8fe9f7ef79d1eded3cba04118b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">831×637 12.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by ldeluigi** [comment]
<p>After I SUBMITTED a review with 2 comments and a body, my workflow triggered 4 times, instead of 3:</p>
<ol>
<li>One time with pull_request_review submitted, with the correct body of the review</li>
<li>One time with pull_request_review_comment created, with the correct body of the first comment</li>
<li>One time with pull_request_review_comment created, with the correct body of the second comment</li>
<li><em>One time with pull_request_review edited, with the correct body of the review, but I wasn’t listening for that event</em></li>
</ol>

**Comment by otter-computer** [comment, marked as answer]
<p>I believe this is caused by a typo in your workflow file. It looks like you’ve used <code>type</code> instead of <code>types</code> for both <code>pull_request_review</code> and <code>pull_request_review_comment</code>. This means that your workflow will listen to all default triggers for these events.</p>
<p>If you change both instances to <code>types</code> then it should only listen for the events specified in the array.</p>
<p>I’m surprised the workflow editor doesn’t flag this as an error or at least show a warning though. I’ll pass this on to the editor team and see if there’s an open issue for it.</p>
